### PR TITLE
CI: Add AKS helm overrides for E2E test

### PR DESCRIPTION
**Pull request by vipul-21**
AKS does not have the E2E integration enabled for AKS. This PR adds `aks` as a valid `CNI_INTEGRATION` attribute. The change includes the helm overrides and the test that should be skipped for AKS.

### CODE REVIEW

**Comment by vipul-21** [comment]
@joestringer  As discussed, the PR is for ginkgo tests that are needed for the AKS.

File: test/helpers/kubectl.go
Comment (joestringer) [inline test/helpers/kubectl.go]: Could you parameterize these `repository` + `tag` references? I think it makes sense to be able to launch the testsuite against target development images like this, but given that the Cilium CNCF project doesn't have control over these Microsoft repositories, it is a bit odd to hardcode these references into the code tree. Also it would be useful to not have to update the version string here every time that we want to run tests against a different version.

In other cases, we have just defined these via environment variables and passed them in as overrides.

File: test/helpers/kubectl.go
Comment (joestringer) [inline test/helpers/kubectl.go]: I'm not sure exactly how this works but if this often varies in the cluster depending on how you provisioned it, then maybe this one should be parameterized as well. This is already parameterized for GKE runs (`grep NATIVE_CIDR` in the test/ dir).

File: test/helpers/kubectl.go
Comment (joestringer) [inline test/helpers/kubectl.go]: nit: These whitespace changes seem unnecessary.

File: stable.txt
Comment (joestringer) [inline stable.txt]: I would suggest not changing this file since it will frequently conflict and change.

**Review by joestringer — CHANGES_REQUESTED** [review]
Thanks for the submission. Just a couple of suggestions mainly around parameterizing options that will often change, plus some other nits here and there.

File: test/helpers/kubectl.go
Comment (vipul-21) [reply test/helpers/kubectl.go]: Sure will update the PR.

File: stable.txt
Comment (vipul-21) [reply stable.txt]: Sure will update the PR.

File: test/helpers/kubectl.go
Comment (vipul-21) [reply test/helpers/kubectl.go]: Sure, make sense. So you want the test case to have a default image and tag, but can be passed through parameters as well. Is that correct ?

File: test/helpers/kubectl.go
Comment (vipul-21) [reply test/helpers/kubectl.go]: Regarding this a well. Make the default value as this and pass it as parameter through cmd line ? 
I saw the GKE code, they are fetching it from the ENV Variable. What should be the ideal way to pass the parameterized this ?

Comment (vipul-21) [reply test/helpers/kubectl.go]: Also, how will this affect the test cases being run in the PR ? ( having a parametric value in the test cases.)

File: test/helpers/kubectl.go
Comment (joestringer) [reply test/helpers/kubectl.go]: Yeah that's what I had in mind. I suspect there's already some way to do this already, because the existing cloud jobs would need to pull the image from somewhere (which is typically from cilium/cilium-ci).

File: test/helpers/kubectl.go
Comment (joestringer) [reply test/helpers/kubectl.go]: Environment variables are usually an easy way to do this, because you can then just set up your own environment with the relevant fields and it will just work. This works for either CI type environments or your local development machine.

I'm not quite sure I understand your concern around parametric values. For the IPv4 Native Routing CIDR, I would expect that if it is configured correctly, all tests pass and if it's not configured correctly then some particular forwarding path may experience problems because e.g. masquerading is misconfigured.

File: stable.txt
Comment (vipul-21) [reply stable.txt]: I think the version got updated to v.1.12.2. So will resolve the conflict.

File: Documentation/contributing/testing/e2e.rst
Comment (joestringer) [inline Documentation/contributing/testing/e2e.rst]: nit, this doesn't need to be a note.

```suggestion
   cluster. You do not need to deploy Cilium in this step, as the End-To-End
   Testing Framework handles the deployment of Cilium.
```

File: stable.txt
Comment (joestringer) [inline stable.txt]: Please drop the changes to this file, there's no need to modify it.

**Review by joestringer — CHANGES_REQUESTED** [review]
Overall the changes look good, couple of minor nits below. Please also drop the merge commit and instead just `git rebase` on top of the latest master branch.

File: Documentation/contributing/testing/e2e.rst
Comment (joestringer) [inline Documentation/contributing/testing/e2e.rst]: I think that the indendation might not be right here either, typically I think we use indendation of 4 spaces for the code below a `.. code-block: ...` since sphinx/rST is opinionated about this. You can use `make render-docs` to see how the page is rendered.

**Review by joestringer — APPROVED** [review]
LGTM, thanks!

**Comment by joestringer** [comment]
cc @cilium/ci-structure

File: test/helpers/utils.go
Comment (christarazi) [inline test/helpers/utils.go:521]: Nit: Seems like this does the same thing as `GKENativeRoutingCIDR()`, maybe consolidate these functions into `NativeRoutingCIDR()` instead?

File: test/k8s/updates.go
Comment (christarazi) [inline test/k8s/updates.go:51]: Nit: No need for extra line

**Review by christarazi — APPROVED** [review]

File: test/helpers/utils.go
Comment (vipul-21) [reply test/helpers/utils.go:521]: I thought of doing so. I am not fully versed with the impact it might have on changing the name `GKENativeRoutingCIDR()` to `NativeRoutingCIDR()`. There is only use one case for that function. Not sure if that'll break something for GKE test cases.

Maybe want to keep GKE and AKS CIDR function segregated?

Comment (christarazi) [reply test/helpers/utils.go:521]: Both of them rely on the `NATIVE_CIDR`env var anyway  so there's no special logic or unexpected impact that I can see.

Comment (joestringer) [reply test/helpers/utils.go:521]: It's configuring the same option, so as long as it's reading the same environment variable I think this should be OK.

**Comment by joestringer** [comment]
The only two remaining pieces of feedback are trivial and not worth blocking this PR on, then there's only CLI / hubble codeowners left but none of these logic changes are related to those codeowner areas. This will only be used outside the CI for now, so I think we're good to merge this as-is.

**Pull request merged**
